### DISCUSSION TITLE
Create ERD programatically

### DISCUSSION THREAD

**Discussion by EslamAshour** [schemacrawler/SchemaCrawler#630] [Q&A]
Hello, 

I need to integrate with schemacrawler to create a new project with Java that generates ERD for any database connection.

can you please help me with the steps I should do or share with me an example of a java program that creates ERD from a DB connection? 

Thanks

**Comment by sualeh** [comment]
@EslamAshour - please take a look at [schemacrawler-examplecode](https://github.com/schemacrawler/SchemaCrawler/tree/master/schemacrawler-examplecode).

**Reply by EslamAshour** [reply]
Hello @sualeh, 

Thanks for your reply, I checked the three examples but I think they don't generate ERD.
below is a screenshot from the output file from the  ExecutableExample

![Capture2](https://user-images.githubusercontent.com/25147996/156314236-ccbae444-949d-4f19-870d-ffa512cf5da1.JPG)

can you please help me with how can I generate image for ERD? 

Thanks

**Comment by sualeh** [comment]
@EslamAshour - please take a look at [DiagramOutputTest.java](https://github.com/schemacrawler/SchemaCrawler/blob/master/schemacrawler-diagram/src/test/java/schemacrawler/integration/test/DiagramOutputTest.java)

**Reply by EslamAshour** [reply]
Hello @sualeh , 

The output file is generated image or HTML but not editable, is there any way to generate an editable file? 

Thanks for your support.

**Comment by sualeh** [comment]
@EslamAshour - please take a look at:

- [How to Generate dbdiagram.io Diagrams for Your Database](https://dev.to/sualeh/how-to-generate-dbdiagram-io-diagrams-for-your-database-431l)
- [How to Generate Mermaid Diagrams for Your Database](https://dev.to/sualeh/how-to-generate-mermaid-diagrams-for-your-database-33bn)
